src/Integration/Binding/DotNetBindingConfigProvider.cs

```diff
@@ -44,83 +45,98 @@ internal class DotNetBindingConfigProvider : IBindingConfigProvider
-
-            // Give up if the quality profile is empty
-            if (ruleSet == null ||
-                ruleSet.Rules.Count == 0 ||
-                ruleSet.Rules.All(rule => rule.Action == RuleAction.None))
+            // Give up if the quality profile is empty - not point in fetching anything else
+            if (!activeRules.Any())
             {
                 this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
                     string.Format(Strings.NoSonarAnalyzerActiveRulesForQualityProfile, qualityProfile.Name, language.Name)));
                 return null;
             }
 
-            // duncanp - WIP - next line will be removed when the RoslynProfileExporter is removed
-            var nugetPackages = roslynProfileExporter.Deployment.NuGetPackages.Select(x => new Core.CSharpVB.NuGetPackageInfo(x.Id, x.Version));
+            // Now fetch the data required for the NuGet configuration
+            var serverProperties = await WebServiceHelper.SafeServiceCallAsync(
+                () => sonarQubeService.GetAllPropertiesAsync(null, cancellationToken), logger);
+            var sonarProperties = serverProperties.ToDictionary(x => x.Key, x => x.Value);
 
-            // Add the NuGet reference, if appropriate (only in legacy connected mode, and only C#/VB)
-            if (!this.nuGetBindingOperation.ProcessExport(language, nugetPackages))
+            // Get the NuGet package info and process it if appropriate (only in legacy connected mode, and only C#/VB)
+            var nugetInfo = nuGetPackageInfoGenerator.GetNuGetPackageInfos(activeRules, sonarProperties);
+            if (!this.nuGetBindingOperation.ProcessExport(language, nugetInfo))
             {
                 return null;
             }
 
-            // Remove/Move/Refactor code when XML ruleset file is no longer downloaded but the proper API is used to retrieve rules
-            UpdateDownloadedSonarQubeQualityProfile(ruleSet, qualityProfile, this.projectName, this.serverUrl);
+            // Finally, fetch the remaining data needed to build the ruleset
+            var inactiveRules = await WebServiceHelper.SafeServiceCallAsync(
+                () => sonarQubeService.GetRulesAsync(false, qualityProfile.Key, cancellationToken), logger);
 
-            return new DotNetBindingConfigFile(ruleSet);
-        }
+            // Create the ruleset
+            var coreRuleset = ruleSetGenerator.Generate(language.ServerLanguage.Key, activeRules.Union(inactiveRules), sonarProperties);
 
-        private void UpdateDownloadedSonarQubeQualityProfile(RuleSet ruleSet, SonarQubeQualityProfile qualityProfile, string projectName, string serverUrl)
-        {
-            ruleSet.NonLocalizedDisplayName = string.Format(Strings.SonarQubeRuleSetNameFormat, projectName, qualityProfile.Name);
+            // Set the name and description
+            coreRuleset.Name = string.Format(Strings.SonarQubeRuleSetNameFormat, projectName, qualityProfile.Name);
```

Comment:
I'm not sure setting the name and description is useful (particularly having the API URL encoded in the description).
It isn't visible in the VS UI anywhere so the user would only see it if the open the file in a text editor.
Also, we need to pass in extra fields just to construct the messages.

However, I've left it as-if for now. Then only difference is that the separator in the description is now a space rather than a line-break. The line-break was being escaped in the file since XML attributes have to be single-line.